src/registries/npm-registry.js

```diff
@@ -139,7 +138,7 @@ export default class NpmRegistry extends Registry {
         await fs.mkdirp(mirrorLoc);
       }
 
-      defaults(this.config, config);
+      this.config = Object.assign({}, config, this.config);
```

Comment:
I don't think it is.  `this.config` is initialized [as an empty object](https://github.com/yarnpkg/yarn/blob/master/src/registries/base-registry.js#L29) when a registry is constructed, and its values are loaded from [a .ini](https://github.com/yarnpkg/yarn/blob/master/src/registries/npm-registry.js#L129) or [a lockfile](https://github.com/yarnpkg/yarn/blob/master/src/registries/yarn-registry.js#L81).
The only case I came come up with is if you have something like
```
const yarnReg = new YarnRegistry();
await yarnReg.loadConfig();
const config = yarnReg.config.
await yarnReg.loadConfig();
// config and yarnReg.config will be different and may have different values if the lockfile has been edited between `yarReg.loadConfig` calls.
```